src/main/java/com/cloudbees/jenkins/support/filter/SensitiveContentFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.jenkins.support.filter;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Filters contents based on names provided by all {@linkplain NameProvider known sources}.
+ *
+ * @see NameProvider
+ * @since TODO
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class SensitiveContentFilter implements ContentFilter {
+
+    public static SensitiveContentFilter get() {
+        return ExtensionList.lookupSingleton(SensitiveContentFilter.class);
+    }
+
+    @Override
+    public @Nonnull String filter(@Nonnull String input) {
+        ContentMappings mappings = ContentMappings.get();
+        String filtered = input;
+        for (ContentMapping mapping : mappings) {
+            filtered = mapping.filter(filtered);
+        }
+        return filtered;
+    }
+
+    @Override
+    public synchronized void reload() {
+        ContentMappings mappings = ContentMappings.get();
+        Set<String> stopWords = mappings.getStopWords();
+        for (NameProvider provider : NameProvider.all()) {
+            provider.names()
+                    .filter(name -> !stopWords.contains(name.toLowerCase(Locale.ENGLISH)))
```

Comment:
After the latest commit, tests are failing because `Computer#getName` sometimes returns "", which makes it into a mapping here and causes all kinds of strangeness.

I would add something like the following to fix the tests:

    diff --git a/src/main/java/com/cloudbees/jenkins/support/filter/SensitiveContentFilter.java b/src/main/java/com/cloudbees/jenkins/support/filter/SensitiveContentFilter.java
    index e0f3660..42893a3 100644
    --- a/src/main/java/com/cloudbees/jenkins/support/filter/SensitiveContentFilter.java
    +++ b/src/main/java/com/cloudbees/jenkins/support/filter/SensitiveContentFilter.java
    @@ -32,6 +32,7 @@ import org.kohsuke.accmod.restrictions.NoExternalUse;
     import javax.annotation.Nonnull;
     import java.util.Locale;
     import java.util.Set;
    +import org.apache.commons.lang.StringUtils;
     
     /**
      * Filters contents based on names provided by all {@linkplain NameProvider known sources}.
    @@ -63,6 +64,7 @@ public class SensitiveContentFilter implements ContentFilter {
             Set<String> stopWords = mappings.getStopWords();
             for (NameProvider provider : NameProvider.all()) {
                 provider.names()
    +                    .filter(name -> !StringUtils.isBlank(name))
                         .filter(name -> !stopWords.contains(name.toLowerCase(Locale.ENGLISH)))
                         .forEach(name -> mappings.getMappingOrCreate(name, original -> ContentMapping.of(original, provider.generateFake())));
             }

Reply:
Whoops! Fixed. There's even a `StringUtils.isNotBlank(name)` which reads well.